### PR TITLE
feat: add Toast component

### PR DESCRIPTION
**Pull request by marboro92**
## Description

Adds a toast component and an example of multiple toasts.

## Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)
- [ ] Documentation only

## Checklist

**Please check all that apply.**

- [x] Storybook updated with examples of new functionality
- [x] Storybook uses variable and realistic data (ex: short and long text)
- [x] Docs updated with correct props and examples
- [x] Updated and reviewed changes to storyshots
- [x] e2e tests added for component interations
- [x] jest tests added for component API that may not be captured with storyshots (change handlers, renderers etc)
- [ ] Accessibility (includes relevant tags, keyboard functionality, colour contrast)

## Before Merging

- [ ] Tested storybook deployment preview
- [ ] Tested docs deployment preview

### CODE REVIEW

File: components/src/Toast/Toast.story.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    };
+    return <ToastWithTrigger />;
+  })
+  .add("multiple toasts example", () => {
+    const MultipleToastsExample = () => {
+      const [currentToast, setCurrentToast] = useState(undefined);
+      const triggerToast = toastName => {
+        setCurrentToast(toastName);
+      };
+      /* Alternate trigger toast fn adds a delay between toasts
```
Comment (marboro92) [inline components/src/Toast/Toast.story.js:34]: I'm not sure about whether this delay is needed. I tried but waiting for the original toast to fade out makes clicking the buttons feel less responsive.. lmk your thoughts you can uncomment it and comment out the other function to try it out.

File: components/src/Toast/Toast.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+};
+
+const FADE_OUT_STYLES = {
+  transition: "opacity 1s linear"
+};
+
+const AnimatedAlert = styled(Alert)(({ visible }) => ({
+  boxShadow: theme.shadows.medium,
+  minWidth: "200px",
+  maxWidth: "600px",
```
Comment (marboro92) [inline components/src/Toast/Toast.js:38]: just using this max width to restrict the characters per line.. I'm not doing anything to enforce the number or lines/max text in the code, I opted for just adding that usage info in docs.
 It seems like that would be a confusing if the component cut off text that was inputted and would make it difficult to deal with longer translations

Comment (nikola-nulogy) [reply components/src/Toast/Toast.js:38]: This makes sense to me.

File: components/src/Toast/Toast.story.js (hunk 1 above)
Comment (nikola-nulogy) [reply components/src/Toast/Toast.story.js:34]: Honestly, I can't tell the difference and I am good with either option.

**Review by nikola-nulogy — CHANGES_REQUESTED** [review]
Title on the doc site it Tooltip

**Review by nikola-nulogy — APPROVED** [review]

**Pull request merged**